# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: steve.langasek@canonical.com-20120128215828-\
#   3a01a4sb68no2ngu
# target_branch: bzr+ssh://bzr.debian.org/bzr/apt/apt/debian-\
#   experimental2
# testament_sha1: 1685d7e4e49b557dc9b9481a3f1a15d2c2fd0b5d
# timestamp: 2012-01-28 14:07:35 -0800
# source_branch: bzr+ssh://bzr.debian.org/bzr/apt/apt/debian-\
#   experimental2
# base_revision_id: egon@debian-devbox-20120124130204-xsn9e788t4dl13xi
# 
# Begin patch
=== modified file 'cmdline/apt-get.cc'
--- cmdline/apt-get.cc	2012-01-20 00:02:36 +0000
+++ cmdline/apt-get.cc	2012-01-28 21:58:28 +0000
@@ -2863,10 +2863,16 @@
 	       if ((BADVER(Ver)) == false)
 	       {
 		  string forbidden;
-		  if (Ver->MultiArch == pkgCache::Version::None || Ver->MultiArch == pkgCache::Version::All);
+		  if (Ver->MultiArch == pkgCache::Version::None || Ver->MultiArch == pkgCache::Version::All)
+		  {
+		     if (colon == string::npos)
+		     {
+			Pkg = Ver.ParentPkg().Group().FindPkg(hostArch);
+		     }
+		  }
 		  else if (Ver->MultiArch == pkgCache::Version::Same)
 		  {
-		     if (colon != string::npos)
+		     if (colon == string::npos)
 			Pkg = Ver.ParentPkg().Group().FindPkg(hostArch);
 		     else if (strcmp(D->Package.c_str() + colon, ":any") == 0)
 			forbidden = "Multi-Arch: same";

=== modified file 'debian/changelog'
--- debian/changelog	2012-01-24 13:02:04 +0000
+++ debian/changelog	2012-01-28 21:58:28 +0000
@@ -1,3 +1,10 @@
+apt (0.8.16~exp13) UNRELEASED; urgency=low
+
+  * Multi-Arch: none build-deps should be DEB_HOST_ARCH, not DEB_BUILD_ARCH.
+    Closes: #646288.
+
+ -- Steve Langasek <vorlon@debian.org>  Sat, 28 Jan 2012 21:23:06 +0000
+
 apt (0.8.16~exp12) experimental; urgency=low
 
   [ Michael Vogt ]

=== modified file 'test/integration/test-bug-632221-cross-dependency-satisfaction'
--- test/integration/test-bug-632221-cross-dependency-satisfaction	2011-09-09 10:35:22 +0000
+++ test/integration/test-bug-632221-cross-dependency-satisfaction	2012-01-28 21:58:28 +0000
@@ -6,43 +6,51 @@
 setupenvironment
 configarchitecture 'amd64' 'armel'
 
-insertinstalledpackage 'build-essential' 'all' '11.5'
+insertinstalledpackage 'build-essential' 'all' '11.5' 'Multi-Arch: foreign'
 
-insertpackage 'unstable' 'doxygen' 'amd64,armel' '1.0'
+insertpackage 'unstable' 'doxygen' 'amd64,armel' '1.0' 'Multi-Arch: foreign'
 insertpackage 'unstable' 'libc6' 'amd64,armel' '1.0' 'Multi-Arch: same'
 insertpackage 'unstable' 'libc6-dev' 'amd64,armel' '1.0' 'Depends: libc6
 Multi-Arch: same'
+insertpackage 'unstable' 'libfwibble1' 'amd64,armel' '1.0' 'Depends: libc6
+Multi-Arch: same'
+insertpackage 'unstable' 'libfwibble-dev' 'amd64,armel' '1.0' 'Depends: libfwibble1'
 insertpackage 'unstable' 'cool' 'amd64,armel' '1.0' 'Multi-Arch: allowed'
 insertpackage 'unstable' 'amdboot' 'amd64' '1.0'
 insertpackage 'unstable' 'foreigner' 'amd64,armel' '1.0' 'Multi-Arch: foreign'
 
-insertsource 'unstable' 'apt' 'any' '0.8.15' 'Build-Depends: doxygen, libc6-dev, libc6-dev:native, cool:any, amdboot:amd64, foreigner'
+insertsource 'unstable' 'apt' 'any' '0.8.15' 'Build-Depends: doxygen, libc6-dev, libc6-dev:native, cool:any, amdboot:amd64, foreigner, libfwibble-dev'
 
 setupaptarchive
 
 testequal 'Reading package lists...
 Building dependency tree...
 The following NEW packages will be installed:
-  amdboot cool doxygen foreigner libc6 libc6-dev
-0 upgraded, 6 newly installed, 0 to remove and 0 not upgraded.
+  amdboot cool doxygen foreigner libc6 libc6-dev libfwibble-dev libfwibble1
+0 upgraded, 8 newly installed, 0 to remove and 0 not upgraded.
 Inst amdboot (1.0 unstable [amd64])
 Inst cool (1.0 unstable [amd64])
 Inst doxygen (1.0 unstable [amd64])
 Inst foreigner (1.0 unstable [amd64])
 Inst libc6 (1.0 unstable [amd64])
 Inst libc6-dev (1.0 unstable [amd64])
+Inst libfwibble1 (1.0 unstable [amd64])
+Inst libfwibble-dev (1.0 unstable [amd64])
 Conf amdboot (1.0 unstable [amd64])
 Conf cool (1.0 unstable [amd64])
 Conf doxygen (1.0 unstable [amd64])
 Conf foreigner (1.0 unstable [amd64])
 Conf libc6 (1.0 unstable [amd64])
-Conf libc6-dev (1.0 unstable [amd64])' aptget build-dep apt -s
+Conf libc6-dev (1.0 unstable [amd64])
+Conf libfwibble1 (1.0 unstable [amd64])
+Conf libfwibble-dev (1.0 unstable [amd64])' aptget build-dep apt -s
 
 testequal 'Reading package lists...
 Building dependency tree...
 The following NEW packages will be installed:
   amdboot cool doxygen foreigner libc6 libc6:armel libc6-dev libc6-dev:armel
-0 upgraded, 8 newly installed, 0 to remove and 0 not upgraded.
+  libfwibble-dev:armel libfwibble1:armel
+0 upgraded, 10 newly installed, 0 to remove and 0 not upgraded.
 Inst amdboot (1.0 unstable [amd64])
 Inst cool (1.0 unstable [amd64])
 Inst doxygen (1.0 unstable [amd64])
@@ -51,6 +59,8 @@
 Inst libc6:armel (1.0 unstable [armel])
 Inst libc6-dev (1.0 unstable [amd64])
 Inst libc6-dev:armel (1.0 unstable [armel])
+Inst libfwibble1:armel (1.0 unstable [armel])
+Inst libfwibble-dev:armel (1.0 unstable [armel])
 Conf amdboot (1.0 unstable [amd64])
 Conf cool (1.0 unstable [amd64])
 Conf doxygen (1.0 unstable [amd64])
@@ -58,34 +68,41 @@
 Conf libc6 (1.0 unstable [amd64])
 Conf libc6:armel (1.0 unstable [armel])
 Conf libc6-dev (1.0 unstable [amd64])
-Conf libc6-dev:armel (1.0 unstable [armel])' aptget build-dep apt -s -a armel
+Conf libc6-dev:armel (1.0 unstable [armel])
+Conf libfwibble1:armel (1.0 unstable [armel])
+Conf libfwibble-dev:armel (1.0 unstable [armel])' aptget build-dep apt -s -a armel
 
 configarchitecture 'armel' 'amd64'
 
 testequal 'Reading package lists...
 Building dependency tree...
 The following NEW packages will be installed:
-  amdboot:amd64 cool doxygen foreigner libc6 libc6-dev
-0 upgraded, 6 newly installed, 0 to remove and 0 not upgraded.
+  amdboot:amd64 cool doxygen foreigner libc6 libc6-dev libfwibble-dev
+  libfwibble1
+0 upgraded, 8 newly installed, 0 to remove and 0 not upgraded.
 Inst amdboot:amd64 (1.0 unstable [amd64])
 Inst cool (1.0 unstable [armel])
 Inst doxygen (1.0 unstable [armel])
 Inst foreigner (1.0 unstable [armel])
 Inst libc6 (1.0 unstable [armel])
 Inst libc6-dev (1.0 unstable [armel])
+Inst libfwibble1 (1.0 unstable [armel])
+Inst libfwibble-dev (1.0 unstable [armel])
 Conf amdboot:amd64 (1.0 unstable [amd64])
 Conf cool (1.0 unstable [armel])
 Conf doxygen (1.0 unstable [armel])
 Conf foreigner (1.0 unstable [armel])
 Conf libc6 (1.0 unstable [armel])
-Conf libc6-dev (1.0 unstable [armel])' aptget build-dep apt -s
+Conf libc6-dev (1.0 unstable [armel])
+Conf libfwibble1 (1.0 unstable [armel])
+Conf libfwibble-dev (1.0 unstable [armel])' aptget build-dep apt -s
 
 testequal 'Reading package lists...
 Building dependency tree...
 The following NEW packages will be installed:
   amdboot:amd64 cool doxygen foreigner libc6:amd64 libc6 libc6-dev:amd64
-  libc6-dev
-0 upgraded, 8 newly installed, 0 to remove and 0 not upgraded.
+  libc6-dev libfwibble-dev:amd64 libfwibble1:amd64
+0 upgraded, 10 newly installed, 0 to remove and 0 not upgraded.
 Inst amdboot:amd64 (1.0 unstable [amd64])
 Inst cool (1.0 unstable [armel])
 Inst doxygen (1.0 unstable [armel])
@@ -94,6 +111,8 @@
 Inst libc6 (1.0 unstable [armel])
 Inst libc6-dev:amd64 (1.0 unstable [amd64])
 Inst libc6-dev (1.0 unstable [armel])
+Inst libfwibble1:amd64 (1.0 unstable [amd64])
+Inst libfwibble-dev:amd64 (1.0 unstable [amd64])
 Conf amdboot:amd64 (1.0 unstable [amd64])
 Conf cool (1.0 unstable [armel])
 Conf doxygen (1.0 unstable [armel])
@@ -101,7 +120,9 @@
 Conf libc6:amd64 (1.0 unstable [amd64])
 Conf libc6 (1.0 unstable [armel])
 Conf libc6-dev:amd64 (1.0 unstable [amd64])
-Conf libc6-dev (1.0 unstable [armel])' aptget build-dep apt -s -a amd64
+Conf libc6-dev (1.0 unstable [armel])
+Conf libfwibble1:amd64 (1.0 unstable [amd64])
+Conf libfwibble-dev:amd64 (1.0 unstable [amd64])' aptget build-dep apt -s -a amd64
 
 configarchitecture 'amd64' 'armel'
 
@@ -111,34 +132,43 @@
 testequal 'Reading package lists...
 Building dependency tree...
 The following NEW packages will be installed:
-  amdboot doxygen libc6 libc6-dev
-0 upgraded, 4 newly installed, 0 to remove and 2 not upgraded.
+  amdboot doxygen libc6 libc6-dev libfwibble-dev libfwibble1
+0 upgraded, 6 newly installed, 0 to remove and 2 not upgraded.
 Inst amdboot (1.0 unstable [amd64])
 Inst doxygen (1.0 unstable [amd64])
 Inst libc6 (1.0 unstable [amd64])
 Inst libc6-dev (1.0 unstable [amd64])
+Inst libfwibble1 (1.0 unstable [amd64])
+Inst libfwibble-dev (1.0 unstable [amd64])
 Conf amdboot (1.0 unstable [amd64])
 Conf doxygen (1.0 unstable [amd64])
 Conf libc6 (1.0 unstable [amd64])
-Conf libc6-dev (1.0 unstable [amd64])' aptget build-dep apt -s
+Conf libc6-dev (1.0 unstable [amd64])
+Conf libfwibble1 (1.0 unstable [amd64])
+Conf libfwibble-dev (1.0 unstable [amd64])' aptget build-dep apt -s
 
 testequal 'Reading package lists...
 Building dependency tree...
 The following NEW packages will be installed:
   amdboot doxygen libc6 libc6:armel libc6-dev libc6-dev:armel
-0 upgraded, 6 newly installed, 0 to remove and 2 not upgraded.
+  libfwibble-dev:armel libfwibble1:armel
+0 upgraded, 8 newly installed, 0 to remove and 2 not upgraded.
 Inst amdboot (1.0 unstable [amd64])
 Inst doxygen (1.0 unstable [amd64])
 Inst libc6 (1.0 unstable [amd64])
 Inst libc6:armel (1.0 unstable [armel])
 Inst libc6-dev (1.0 unstable [amd64])
 Inst libc6-dev:armel (1.0 unstable [armel])
+Inst libfwibble1:armel (1.0 unstable [armel])
+Inst libfwibble-dev:armel (1.0 unstable [armel])
 Conf amdboot (1.0 unstable [amd64])
 Conf doxygen (1.0 unstable [amd64])
 Conf libc6 (1.0 unstable [amd64])
 Conf libc6:armel (1.0 unstable [armel])
 Conf libc6-dev (1.0 unstable [amd64])
-Conf libc6-dev:armel (1.0 unstable [armel])' aptget build-dep apt -s -a armel
+Conf libc6-dev:armel (1.0 unstable [armel])
+Conf libfwibble1:armel (1.0 unstable [armel])
+Conf libfwibble-dev:armel (1.0 unstable [armel])' aptget build-dep apt -s -a armel
 
 configarchitecture 'armel' 'amd64'
 
@@ -148,36 +178,45 @@
 The following packages will be REMOVED:
   cool:amd64
 The following NEW packages will be installed:
-  amdboot:amd64 cool doxygen libc6 libc6-dev
-0 upgraded, 5 newly installed, 1 to remove and 1 not upgraded.
+  amdboot:amd64 cool doxygen libc6 libc6-dev libfwibble-dev libfwibble1
+0 upgraded, 7 newly installed, 1 to remove and 1 not upgraded.
 Remv cool:amd64 [0.5]
 Inst amdboot:amd64 (1.0 unstable [amd64])
 Inst cool (1.0 unstable [armel])
 Inst doxygen (1.0 unstable [armel])
 Inst libc6 (1.0 unstable [armel])
 Inst libc6-dev (1.0 unstable [armel])
+Inst libfwibble1 (1.0 unstable [armel])
+Inst libfwibble-dev (1.0 unstable [armel])
 Conf amdboot:amd64 (1.0 unstable [amd64])
 Conf cool (1.0 unstable [armel])
 Conf doxygen (1.0 unstable [armel])
 Conf libc6 (1.0 unstable [armel])
-Conf libc6-dev (1.0 unstable [armel])' aptget build-dep apt -s
+Conf libc6-dev (1.0 unstable [armel])
+Conf libfwibble1 (1.0 unstable [armel])
+Conf libfwibble-dev (1.0 unstable [armel])' aptget build-dep apt -s
 
 testequal 'Reading package lists...
 Building dependency tree...
 The following NEW packages will be installed:
   amdboot:amd64 doxygen libc6:amd64 libc6 libc6-dev:amd64 libc6-dev
-0 upgraded, 6 newly installed, 0 to remove and 2 not upgraded.
+  libfwibble-dev:amd64 libfwibble1:amd64
+0 upgraded, 8 newly installed, 0 to remove and 2 not upgraded.
 Inst amdboot:amd64 (1.0 unstable [amd64])
 Inst doxygen (1.0 unstable [armel])
 Inst libc6:amd64 (1.0 unstable [amd64])
 Inst libc6 (1.0 unstable [armel])
 Inst libc6-dev:amd64 (1.0 unstable [amd64])
 Inst libc6-dev (1.0 unstable [armel])
+Inst libfwibble1:amd64 (1.0 unstable [amd64])
+Inst libfwibble-dev:amd64 (1.0 unstable [amd64])
 Conf amdboot:amd64 (1.0 unstable [amd64])
 Conf doxygen (1.0 unstable [armel])
 Conf libc6:amd64 (1.0 unstable [amd64])
 Conf libc6 (1.0 unstable [armel])
 Conf libc6-dev:amd64 (1.0 unstable [amd64])
-Conf libc6-dev (1.0 unstable [armel])' aptget build-dep apt -s -a amd64
+Conf libc6-dev (1.0 unstable [armel])
+Conf libfwibble1:amd64 (1.0 unstable [amd64])
+Conf libfwibble-dev:amd64 (1.0 unstable [amd64])' aptget build-dep apt -s -a amd64
 
 

# Begin bundle
IyBCYXphYXIgcmV2aXNpb24gYnVuZGxlIHY0CiMKQlpoOTFBWSZTWVZta2IAB3tfgGAwWf///3/3
3wq//9//YAx5ufXNl0ASEHQNEF2aAUVQAAAAHGjJkYRiAYTQYBNBoGTJoyZDCAwko0CZGiNMo1PU
0jGptQ0DQGgAAaANA0mCUxpTT9U0BoAAAAAAAAPU0AlT0UGoymjEYmjQDQAAAAA0AAHGjJkYRiAY
TQYBNBoGTJoyZDCAwVJImIAJoDSYmQEyYkwU8UabJGnpD0niakwxBgAxAXhcrybp/tNmG0saWcMf
BjV8GaeaIpmzOS1sHRUihqbp7XpQmdkISIWq8wW4EPtL56ADMwUJueheFagMApAdsaoBMTP6dW4+
I5rU8h5AZgikqZJFQhO0ZYlmVokoxgvQhB7wK4gZQEaZUAMCgg+reIlQIYYYYRbZJxhgQwhhFMu1
fru19poHkfnOlVUkVVUKqqP7nuk/X9JMhQjOVieeD3FIsJFlF1FW6zIdSTMXlGemEHQFBOK1ERTS
ZsShJGAsqCAZzyEeqCOYjBZMxhXY6TRICtVqa0lEVpEWSNRhVUBIwVkRIJYXz9C2Jjk3VPj1/R5f
5+vBEf4YHD97x3jyGDtX7BTbd80uTMqrZ4lfqZqhi2pCkjbpK9a04EsxPpsK+QzgqTOLRrjiENB3
HHkccYeCYcQiNgccsDDFg3+aq02eWdqeTvrE5uO/v1Ha9wsmn1L1wwzlnJAtAPdp0w3e7ft7eMqq
3e7jwQUHHZ7SradtREpPUNb90/VNerwtvz+Bp2yrY448+R6xqIvo9PNYkVXHR+pajz0J81D7Xfj5
6UjZSc23r2t/d/VfPfo+l1ZZ3Y95RHMjo1PhG7Sb3Z7b4dVTVLlFFKioqhSitZdgwRcs52bIc87X
E8d2kdD252sRS3PlV8xWstzLnc47nmgcbY8Uji9Gjifffz4WtaeVqbG/6ZZlCdFI9VKrxqHFUjAR
9ApLlTfvZvl5aWlT77Iz8SrSRK6/oRcbD5PdmXo8KUxLRje5mmGbwpWnLPnZmGlxurPmaGSjMqxV
o70vhkaS/9lvIxr3dAeK7gvhABlLF5vMhlnCJUoXmI2ARun1me4ZfXtpckTKLzHjDlz2JlBF8l1E
DMdQwI76RN2axWZYi1dTpz1v4vnuODi6tDUxfBuw3G7HbyFs6uZ0rl21k2O69rK0o29O692q92Vn
PttpaYcJx35YPXjZ3VoN2wmY4nOjhh1OZk2dd7la/16wwZaGqrqthsccZyvWZueHeXXWhKEeExGA
/ALlCiB0JzEPY0fi7EgtlUpFQOnn69MlMMpci7Fo45LBx2PgXkQxeIoUORIYRMEPFVD9LR62UYoP
9LkzvlMfgEdwix1ZAt8tHK2b3ZycFiaowxCSnAaSnGBLm+LEOsgTcVH7wIFqcxDFDpwdGAQIPqZQ
Hlk8DGLnhXvvBgLQUgZANSoG47ELqzowmFUrgg7kDEScnGA9/MSOY4mZuImgsRBFBcxY6hxkwzeQ
KjQ8BkI0AnltpbOF6xc9vUeO0LcDwhMk5gzpjWCua5FSbbdXYYzUHm5g+iSjrZm1L0zIxLnlNhHi
0ullro/K8gk96SaAVDgSMcN+NUZjnOl00HlDAEbeX6CxAQXPF1Z9OXN9hbHEbmIe802MkoZcB0BF
jA2NytK7wqylGjDF3GMcRhGY1qjLocHlca0dV1LjOdnBnNNKYQ3s4tNzm18NGlh0yKx3b46JoZaa
tCUZBRhDbGr6EZavfMENqVLCM2cxhiTQ90p8MIBB0Aw4ltS0XGRsshjWJeXEQ8K3VsovdVxCHSLY
cd3RbgZ1vhhSAzPNS1+AcCA4yYe80stXb4T4qVK9e0z46Wm0vtN8KNTpdjl11tUyeKinnUrjcuUp
SimhryWTX2OhWnA2rkz6mS8U0KwSkzwz6FPh+WwnsnwTkfUXuLXVSjw6KqVapKwPZ4vSdfqfyjvT
4EVKaY8G8/ry8XDRGaG5xlUpKqrMY+P6G18sIFHkH6laD+YW/m3tn5Bc/syAMfQXeoiDIfrL7SL8
pR2Y2X5B7r4Ka0AqjWThGOdSCAga+4IQoIpSlJSlKVVVSnglk+eHvUfCfTC+Zp+rvffDWeyH5OSp
VYw6aeNlKpHswuHpOHxiOkQXyfLwPMItuenFCRufFl7B1ith9IA6Y2XePkcWMLfQ+j8JJK9+5GZt
eeSXz/KF0aORi/W/PJNSw/lPQkctkkmrX8cRd9iw/n5/FP0nJ+9E2wxaUn5zoSPuhOv56Mevr/p8
3pdD2vZPa9udFplmeX443vAiZsyxGfNg18XtwInsTtJGciVqj8TqYQ/t3XPU/nbUTx/J6BHM3ben
33XL2DnYODHAjoZLX2nvcJJP/pJMmf73Syz9XR19mS87lSRq9md0PM57teC0Wy+tjtgdax28GWWS
Pcsu0WC5JQ+Li7HXy3uxq5/O7Wt5JDdJh0Y6AgcZiH52XYVR37nYHW7cAobVtihRchI47ejr9qZ+
Yz9yMx4Vm1TM/t7Oqpqemdwj1eVw+GhmP725XGbFZEskUh/xr7mq8nB7c2Fa7edwdR2PQ9OrmfWc
Dbn8qo2qtcYw782XFNaz7b+vIYAhxZjxcXOqcuWLG8838seQZ9ZmjxoFskGXbYro87a+1RikdOgU
6aN9eJdfJfr8yXC6HwhXnTlPgSd6VDwcjw2iToMjLZ50DQ7qvHEhGIhbVIXQmzeih6P4vRk9OmYC
eudPqfFS11d3rveVvUu477171pqT6z3nMaa21Cx4kT7r+FnpnTtWc6ObFmwWhN0krkl93SOT53CM
55grU/ZNKOhJNpK0U4U86nP9GD5YduaIZRqVO5vR23Tj1pUNhE+Ah90N8kl9Mcn+yOF1z5zk1O79
mCbo7XWk85MTFlD9+e25aLxmr7OXk7GDNg7Sc63tRf5LtjA6/U496OedZSRUbH4TxfF5oZu9GKT1
yo1dtUnpVCXJrKbU8lGCyLhjSR9SR3s6T0CKKSSipEoxh0wPrqEqian49q0MlTRpnh/u/7+ZNTKT
jbiMoHyZ40hKhHY3nnR4MW0YsE6WN2aZlqls63VdarylKYFSLQqRxMrmWCwsixmjKWwFwl5mRPsR
dJJ8za0o+1GTGSSkaX1aI/MOJIsjWKqSqFSYvR6RzI46dm46EhZdntUjmLfxC5k/53OQl0kmc741
TZshUeD2Wj03pTYwnpaHBQosu7fMMNRm5J0w6vji13w6sLpbkooZkzHfsZwwbty7JjORFY0tUmdT
kF8Nsi3+MvQypKhUMcPDKNYwkml6FyXs+WEPXvZkuhFrtzlP4xhjXoWUopNYwXr3Xw41e6JhZqwb
c/uR4sGZuhK5tUa2HcvZ6K4GlmPMpLjXNtRWDkXaNilagtarREsV/R8Va4pYqW+CobFc0OGdnFba
4BQzZrlmlg932fMZTpTyTQ0Z44CxI7ahdRdK6buLjxr/WYJHEaXAWtGpss0aXErzYTEZ4YJrwly6
1Cud6N/CHw7Nl3nk/Gom963vk3CORfYN1rc3Z2wS8b7JHMlVSPN4s2tneithE9SMh1acjprzJ3TD
N3/iXeJQ7qeGdO920oU53TaPqRa0bNSfTp2Rr1vs5/3q1fh/+LuSKcKEgrNrWxA=
